rust/osp/src/response.rs

```diff
@@ -344,36 +344,25 @@ impl From<&ScanResult> for models::ResultType {
             ("HOST_START", ResultType::Log) => models::ResultType::HostStart,
             ("HOST_END", ResultType::Log) => models::ResultType::HostEnd,
             ("DEADHOST", ResultType::Log) => models::ResultType::DeadHost,
-            (_, ResultType::Log) =>
-            // is either set on nasl-script as proto or on openvas when
-            // a host is dead
-            // we need to clarify if we transform the host detail from
-            // xml to json as map[string]string or if we just keep it as is
-            // and let the user handle it
-            {
-                if sr.port == "general/Host_Details" {
-                    models::ResultType::HostDetail
-                } else {
-                    models::ResultType::Log
-                }
-            }
+            ("Host Details", ResultType::Log) => models::ResultType::HostDetail,
+            (_, ResultType::Log) => models::ResultType::Log,
             (_, ResultType::Alarm) => models::ResultType::Alarm,
             (_, ResultType::Error) => models::ResultType::Error,
-            // hostdetails are sent via log messages
+            // host details are sent via log messages
             (_, ResultType::HostDetail) => unreachable!(),
         }
     }
 }
 
-#[derive(Deserialize, Debug)]
-struct Detail {
-    name: String,
-    value: String,
+#[derive(Deserialize, Debug, Default)]
+struct HostDetail {
+    detail: Vec<models::Detail>,
 }
 
-#[derive(Deserialize, Debug)]
-struct HostDetail {
-    detail: Vec<Detail>,
+impl HostDetail {
+    pub fn extract(&self) -> Option<models::Detail> {
+        self.detail.get(0).cloned()
```

Comment:
should be `first` instead of `get(0)`.

Reply:
Also wouldn't a Option<&models::Detail> be sufficient as it is stored on the HostDetail anyway?

Reply:
`HostDetail` will only live within the function it is used in, the details on the other hand are used for the generated result, so we need to clone it either here or after the `extract()` call